### DISCUSSION TITLE
Any changes in logic for "on-demand" revalidation in Next 13 "app" version?

### DISCUSSION THREAD

**Discussion by andre-mr** [vercel/next.js#43272] [App Router]
Hi.

It seems to be a basic question, but I'm confused and didn't find a clear question yet.

We all know current Next 12.1 / 12.2 on-demand revalidate feature, but with new cache server components way, I found this info:
https://nextjs.org/blog/next-13
```
// This request should be cached until manually invalidated.
// Similar to `getStaticProps`.
// `force-cache` is the default and can be omitted.
fetch(URL, { cache: 'force-cache' });

// This request should be refetched on every request.
// Similar to `getServerSideProps`.
fetch(URL, { cache: 'no-store' });

// This request should be cached with a lifetime of 10 seconds.
// Similar to `getStaticProps` with the `revalidate` option.
fetch(URL, { next: { revalidate: 10 } });
``` 

Ok, but why not mention on-demand revalidate? This makes me believe this is still done via "pages/api/jsfile" call, with that handler using res.revalidate("path"), and is going to be this way for Next 13... or did I misunderstood?

I mean... what is the expected way for "manually invalidate" a cached/static page?

**Comment by icyJoseph** [comment]
I tested this on this page:

```tsx
export default async function Page() {
  const data = await new Promise<number>((next) => {
    setTimeout(next, 1_000, new Date().getTime());
  });

  return (
    <section>
      <h2>Stale Page - {data}</h2>
    </section>
  );
}
```

And it works when calling a revalidate endpoint that takes care of the page.


https://user-images.githubusercontent.com/21013447/203505606-467d24ca-3f05-4651-ba1b-044ca342b048.mov

Do remember that, `appDir` and the `new docs` are all in beta stages. Good to point out that `on-demand` revalidation is missing from the new docs though.
